mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/search/ElasticSearchSearcher.java

```diff
@@ -92,6 +97,27 @@ public ElasticSearchSearcher(Client client, QueryConverter queryConverter, int s
             .orElse(pairStream);
     }
 
+    public long countMessages(Collection<MailboxId> mailboxIds, SearchQuery query) {
```

Comment:
This SHOULD be part of the **search** method returned result (computation costs)